src/pages/ReplaceServer/PrepareController.js

```diff
@@ -0,0 +1,180 @@
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+
+import React from 'react';
+import { translate } from '../../localization/localize.js';
+import { STATUS } from '../../utils/constants.js';
+import BaseUpdateWizardPage from '../BaseUpdateWizardPage.js';
+import { PlaybookProgress } from '../../components/PlaybookProcess.js';
+import { ErrorBanner } from '../../components/Messages.js';
+import { fetchJson, postJson, deleteJson } from '../../utils/RestUtils.js';
+
+class PrepareController extends BaseUpdateWizardPage {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      overallStatus: STATUS.UNKNOWN, // overall status of entire playbook and commit
+      invalidMsg: '',
+    };
+  }
+
+  setNextButtonDisabled = () => this.state.overallStatus != STATUS.COMPLETE;
+
+  updatePageStatus = (status) => {
+    this.setState({overallStatus: status});
+    if (status === STATUS.FAILED) {
+      this.setState({invalidMsg: translate('server.replace.prepare.failure')});
+    }
+  }
+
+  renderPlaybookProgress () {
+    const steps = [
+      {
+        label: translate('deploy.progress.commit'),
+        playbooks: ['commit']
+      },
+      {
+        label: translate('deploy.progress.config-processor-run'),
+        playbooks: ['config-processor-run.yml']
+      },
+      {
+        label: translate('deploy.progress.ready-deployment'),
+        playbooks: ['ready-deployment.yml']
+      },
+      {
+        label: translate('server.deploy.progress.rm-cobbler'),
+        playbooks: ['rm-cobbler']
+      },
+      {
+        label: translate('server.deploy.progress.rm-known-host'),
+        playbooks: ['known-hosts']
+      },
+      {
+        label: translate('server.deploy.progress.cobbler-deploy'),
+        playbooks: ['cobbler-deploy.yml']
+        // TODO:
+        //    cobbler-deploy *might* prompt for a password, although the logs on the QE system
+        //       ardana@10.84.81.17 indicate that none was supplied and it succesfully ran. Hmmmm
+      },
+    ];
+
+    const playbooks = [
+      {
+        name: 'commit',
+        action: ((logger) => {
+          const commitMessage = {'message': 'Committed via Ardana Installer'};
+          return postJson('/api/v1/clm/model/commit', commitMessage)
+            .then((response) => {
+              logger('Model committed\n');
+            })
+            .catch((error) => {
+              const message = translate('update.commit.failure', error.toString());
+              logger(message+'\n');
+              throw new Error(message);
+            });
+        }),
+      },
+      {
+        name: 'config-processor-run',
+      },
+      {
+        name: 'ready-deployment',
+      },
+      {
+        name: 'rm-cobbler',
```

Comment:
Just wondering ...if it is possible to make action step reusable ? Since I need to do the same.

Reply:
That's a good idea. I'll move some of these steps out to another class in another PR